### PR TITLE
Refactor1

### PR DESCRIPTION
**Pull request by syko240**
Summery:
- Changed return types for db functions to return responses, better handling for exceptions
- Docker changes
- some other naming changes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,18 @@
-FROM ubuntu:22.04 as builder
+FROM ubuntu:22.04 AS builder
```
Comment (RyuTribal) [inline Dockerfile:1]: Very good, I think that is one of the most important changes that you've made

**Review by RyuTribal — CHANGES_REQUESTED** [review]
I like this so far. Good use of templated structs to handle errors instead of needless try and catches. 

THis whole thing of segmenting database functionality that belongs to one .h file across multiple .cpp files has to stop though. I never understood the point of it except that you have to jump around more files which takes longer than searching a bigger cpp file. For people who haven't worked on the code base looking in, it's hard to tell which of the files contain the function definition (which are all stored in one .h file). Please changes this. Hopping between files is usually more time consuming than looking at a bigger file

**Comment by RyuTribal** [comment]
Or even better, just make a singleton type of header file for all the transactions that uses the db connection, just dont have multiple definitions spread across multiple cpp files. One header for one cpp file, thats the rule

File: Dockerfile (hunk 1 above)
Comment (syko240) [reply Dockerfile:1]: @RyuTribal upper case standard lil bro

Comment (RyuTribal) [reply Dockerfile:1]: Bro I'm telling you, this change in code, literally changed my whole perception on life

Comment (syko240) [reply Dockerfile:1]: [I am wholeheartedly aligned, my spirit echoing with boundless affirmation, a conviction swelling beyond the measure of certainty—one hundred and one beats in favor of yours truly. For it is not mere agreement that I extend, but a vow, a commitment that stretches beyond words, as if the stars themselves nod in accord. My support is a river unending, flowing deeper and truer than numbers can ever capture, a testament to the faith that lives within.](https://tenor.com/view/funny-gif-23169978)

**Comment by syko240** [comment]
> I like this so far. Good use of templated structs to handle errors instead of needless try and catches.
> 
> THis whole thing of segmenting database functionality that belongs to one .h file across multiple .cpp files has to stop though. I never understood the point of it except that you have to jump around more files which takes longer than searching a bigger cpp file. For people who haven't worked on the code base looking in, it's hard to tell which of the files contain the function definition (which are all stored in one .h file). Please changes this. Hopping between files is usually more time consuming than looking at a bigger file

fair


> Or even better, just make a singleton type of header file for all the transactions that uses the db connection, just dont have multiple definitions spread across multiple cpp files. One header for one cpp file, thats the rule



I will look into it 👍

File: Dockerfile (hunk 1 above)
Comment (mohammed-shakir) [reply Dockerfile:1]: Can you guys stfu?

Comment (RyuTribal) [reply Dockerfile:1]: He is not seeing the light

**Review by RyuTribal — APPROVED** [review]
LGTM!

**Comment by mohammed-shakir** [comment]
(i got tired, there are other places where it should be snake_case)

**Comment by syko240** [comment]
> (i got tired, there are other places where it should be snake_case)

sure, but I dont think it really matters, its more important to distinguish member variables from local or parameter variables.

**Pull request merged**